src/main/java/net/neoforged/neoforge/common/CommonHooks.java

```diff
@@ -238,12 +241,32 @@ public static LivingChangeTargetEvent onLivingChangeTarget(LivingEntity entity,
         return event;
     }
 
-    public static boolean onLivingAttack(LivingEntity entity, DamageSource src, float amount) {
-        return entity instanceof Player || !NeoForge.EVENT_BUS.post(new LivingAttackEvent(entity, src, amount)).isCanceled();
+    /**
+     * Creates and posts an {@link EntityInvulnerabilityCheckEvent}. This is invoked in
+     * {@link Entity#isInvulnerableTo(DamageSource)} and returns a post-listener result
+     * to the invulnerability status of the entity to the damage source.
+     *
+     * @param entity  the entity being checked for invulnerability
+     * @param source  the damage source being applied for this check
+     * @param isInvul whether this entity is invulnerable according to preceding/vanilla logic
+     * @return if this entity is invulnerable
+     */
+    public static boolean onEntityInvulnerabilityCheck(Entity entity, DamageSource source, boolean isInvul) {
+        return NeoForge.EVENT_BUS.post(new EntityInvulnerabilityCheckEvent(entity, source, isInvul)).isInvulnerable();
     }
 
-    public static boolean onPlayerAttack(LivingEntity entity, DamageSource src, float amount) {
-        return !NeoForge.EVENT_BUS.post(new LivingAttackEvent(entity, src, amount)).isCanceled();
+    /**
+     * Called after invulnerability checks in {@link LivingEntity#hurt(DamageSource, float)},
+     * this method creates and posts the first event in the LivingEntity damage sequence,
+     * {@link LivingIncomingDamageEvent}, FOR NON-PLAYER entities.
```

Comment:
> FOR NON-PLAYER entities

Is this still the case after we removed the player-specific overrides?

Reply:
It is not.  Fixed.